Fixes postnl _cleanupString methods, previously when string was cut after maxlength, they used original string for this,
so when customer had for example company name longer than 35 character, but included ampersand - it broke xml

diff --git a/Model/Processor/Common.php b/Model/Processor/Common.php
index 594cf29..52b91c9 100644
--- a/Model/Processor/Common.php
+++ b/Model/Processor/Common.php
@@ -65,8 +65,8 @@ class Common {
     protected function _cleanupString($string, $maxLength = 0)
     {
         $trimmed = trim(preg_replace('#\s+#us', ' ', str_replace($this->_getBadCharacters(), '', $string)));
-        if ($maxLength && mb_strlen($string, 'UTF-8') > $maxLength)
-            $trimmed = mb_substr($string, 0, $maxLength, 'UTF-8');
+        if ($maxLength && mb_strlen($trimmed, 'UTF-8') > $maxLength)
+            $trimmed = mb_substr($trimmed, 0, $maxLength, 'UTF-8');
         return $trimmed;
     }
 
diff --git a/Model/Processor/Order.php b/Model/Processor/Order.php
index d9b5ac8..88f6696 100644
--- a/Model/Processor/Order.php
+++ b/Model/Processor/Order.php
@@ -161,8 +161,8 @@ class Order extends Common {
     protected function _cleanupString($string, $maxLength = 0)
     {
         $trimmed = trim(preg_replace('#\s+#us', ' ', str_replace($this->_getBadCharacters(), '', $string)));
-        if ($maxLength && mb_strlen($string, 'UTF-8') > $maxLength)
-            $trimmed = mb_substr($string, 0, $maxLength, 'UTF-8');
+        if ($maxLength && mb_strlen($trimmed, 'UTF-8') > $maxLength)
+            $trimmed = mb_substr($trimmed, 0, $maxLength, 'UTF-8');
         return $trimmed;
     }
 
